### ISSUE TITLE
1.17 support?

### ISSUE THREAD

**Issue by SkytAsul** [Pante/chimera#290]
**Is your feature request related to a problem? Please describe.**
Currently chimera commons do not support 1.17 as it uses hardcoded NMS.

**Describe the solution you'd like**
I'd like a 1.17 support.

**Comment by Pante** [comment]
We're in the midst of updating Chimera to support 1.17. A snapshot release (5.0.0-SNAPSHOT) will be available shortly (within the day or two). A full release however, is going to take slightly longer as we are still testing `Typist`. At the moment, the only issue holding back the snapshot release is figuring out how to host the different mapped version of Spigot.

**Comment by SkytAsul** [comment]
I can understand you, the NMS mappings update is somehow complicated to manage.
I got this error with the 5.0.0-SNAPSHOT on 1.17:
```
org.bukkit.command.CommandException: Unhandled exception executing 'summonlaser e -79.0 66.0 105.0 -79.0 66.0 105.0 1 ~ ~ ~1' in com.karuslabs.commons.command.dispatcher.DispatcherCommand(summonlaser)
        at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:156) ~[spigot-1.17.jar:3151-Spigot-101ebb6-68d4031]
        at org.bukkit.craftbukkit.v1_17_R1.CraftServer.dispatchCommand(CraftServer.java:764) ~[spigot-1.17.jar:3151-Spigot-101ebb6-68d4031]
        at net.minecraft.server.network.PlayerConnection.handleCommand(PlayerConnection.java:1944) [spigot-1.17.jar:3151-Spigot-101ebb6-68d4031]
        at net.minecraft.server.network.PlayerConnection.a(PlayerConnection.java:1783) [spigot-1.17.jar:3151-Spigot-101ebb6-68d4031]
        at net.minecraft.server.network.PlayerConnection.a(PlayerConnection.java:1764) [spigot-1.17.jar:3151-Spigot-101ebb6-68d4031]
        at net.minecraft.network.protocol.game.PacketPlayInChat.a(PacketPlayInChat.java:46) [spigot-1.17.jar:3151-Spigot-101ebb6-68d4031]
        at net.minecraft.network.protocol.game.PacketPlayInChat.a(PacketPlayInChat.java:1) [spigot-1.17.jar:3151-Spigot-101ebb6-68d4031]
        at net.minecraft.network.protocol.PlayerConnectionUtils.lambda$0(PlayerConnectionUtils.java:30) [spigot-1.17.jar:3151-Spigot-101ebb6-68d4031]
        at net.minecraft.network.protocol.PlayerConnectionUtils$$Lambda$6008/0x00000000f9cdf508.run(Unknown Source) [spigot-1.17.jar:3151-Spigot-101ebb6-68d4031]
        at net.minecraft.server.TickTask.run(SourceFile:18) [spigot-1.17.jar:3151-Spigot-101ebb6-68d4031]
        at net.minecraft.util.thread.IAsyncTaskHandler.executeTask(SourceFile:151) [spigot-1.17.jar:3151-Spigot-101ebb6-68d4031]
        at net.minecraft.util.thread.IAsyncTaskHandlerReentrant.executeTask(SourceFile:23) [spigot-1.17.jar:3151-Spigot-101ebb6-68d4031]
        at net.minecraft.util.thread.IAsyncTaskHandler.executeNext(SourceFile:125) [spigot-1.17.jar:3151-Spigot-101ebb6-68d4031]
        at net.minecraft.server.MinecraftServer.bg(MinecraftServer.java:1125) [spigot-1.17.jar:3151-Spigot-101ebb6-68d4031]
        at net.minecraft.server.MinecraftServer.executeNext(MinecraftServer.java:1118) [spigot-1.17.jar:3151-Spigot-101ebb6-68d4031]
        at net.minecraft.util.thread.IAsyncTaskHandler.executeAll(SourceFile:110) [spigot-1.17.jar:3151-Spigot-101ebb6-68d4031]
        at net.minecraft.server.MinecraftServer.sleepForTick(MinecraftServer.java:1101) [spigot-1.17.jar:3151-Spigot-101ebb6-68d4031]
        at net.minecraft.server.MinecraftServer.x(MinecraftServer.java:1031) [spigot-1.17.jar:3151-Spigot-101ebb6-68d4031]
        at net.minecraft.server.MinecraftServer.lambda$0(MinecraftServer.java:307) [spigot-1.17.jar:3151-Spigot-101ebb6-68d4031]
        at net.minecraft.server.MinecraftServer$$Lambda$3850/0x00000000f5111e88.run(Unknown Source) [spigot-1.17.jar:3151-Spigot-101ebb6-68d4031]
        at java.lang.Thread.run(Thread.java:853) [?:?]
Caused by: java.lang.ExceptionInInitializerError
        at com.karuslabs.commons.command.dispatcher.Exceptions.<clinit>(Exceptions.java:62) ~[?:?]
        at com.karuslabs.commons.command.dispatcher.DispatcherCommand.execute(DispatcherCommand.java:91) ~[?:?]
        at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:149) ~[spigot-1.17.jar:3151-Spigot-101ebb6-68d4031]
        ... 20 more
Caused by: java.lang.NoSuchFieldException: No such field: net.minecraft.commands.CommandDispatcher.LOGGER(org.apache.logging.log4j.Logger)
        at java.lang.invoke.MethodHandles$Lookup.findVarHandleCommon(MethodHandles.java:1761) ~[?:?]
        at java.lang.invoke.MethodHandles$Lookup.findStaticVarHandle(MethodHandles.java:1741) ~[?:?]
        at com.karuslabs.commons.command.dispatcher.Exceptions.<clinit>(Exceptions.java:59) ~[?:?]
        at com.karuslabs.commons.command.dispatcher.DispatcherCommand.execute(DispatcherCommand.java:91) ~[?:?]
        at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:149) ~[spigot-1.17.jar:3151-Spigot-101ebb6-68d4031]
        ... 20 more
        ```

**Comment by Pante** [comment]
After some investigation, we have found the cause to be attempting to retrieve a `VarHandle` for a NMS field that was remapped. Interestingly it wasn't caught in our unit tests since we use Mojang's names during testing, I guess it's something that we need to look out for in the future. We've since fixed the issue in the latest release.

**Comment by Pante** [comment]
`5.0.0` has been released and supports 1.17
